### PR TITLE
[CU-8688vgt4y] Prevents edge caching hidden preferences endpoint

### PR DESCRIPTION
**Pull request by manuelurenah**
Note ⚠️ : to test bug in prod [ ] Disable cache option in Browser network tab should be UNCHECKED

How to test BUG 🐛 

1. Go to /user/:userId/images?section=reactions
2. Hide an image
3. Log out
4. Log in again
5. When you are logged in, you should see recently hidden image

We noticed call `/api/trpc/hiddenPreferences.getHidden` is `Disk cache` in the size column in Network tab

After this solution 
1. Go to /user/:userId/images?section=reactions
2. Hide an image
3. Log out
4. Log in again
5. Hidden image should NOT appear 
 
Call `/api/trpc/hiddenPreferences.getHidden` SHOULT NOT be `Disk cache`

### CODE REVIEW

**Comment by JustMaier** [comment]
Task linked: [CU-8688vgt4y [Ticket: 1956] Displays images on the "My Reactions" page](https://app.clickup.com/t/8688vgt4y)

File: src/pages/api/auth/[...nextauth].ts
Comment (fxisco) [inline src/pages/api/auth/[...nextauth].ts:1]: 🧹

File: src/server/services/user-preferences.service.ts
Comment (fxisco) [inline src/server/services/user-preferences.service.ts:1]: 🧹

**Review by fxisco — APPROVED** [review]
tested with cases reported by ticket! issue related to cache hidden images were fixed ✅

**Pull request merged**